metacatalog_api/db.py

```
@@ -489,3 +489,96 @@ def add_datasource(session: Session, entry_id: int, datasource: models.Datasourc
+        sql = sql.where(models.EntryGroupTable.title == title)
+    
+    group = session.exec(sql).first()
+    if group is None:
+        return None
+    if with_metadata:
+        return models.EntryGroupWithMetadata.model_validate(group)
+    else:
+        return models.EntryGroup.model_validate(group)
+
+
+def add_group(session: Session, title: str, description: str, type: str, entry_ids: list[int] = []) -> models.EntryGroup:
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)